src/identity.cc

```diff
@@ -680,175 +673,178 @@ TRITONBACKEND_ModelInstanceExecute(
-          TRITONSERVER_LOG_ERROR,
-          (std::string("request ") + std::to_string(r) +
-           ": failed to read input or requested output name, error response "
-           "sent")
-              .c_str());
-      continue;
-    }
-
-    TRITONSERVER_DataType input_datatype;
-    const int64_t* input_shape;
-    uint32_t input_dims_count;
-    uint64_t input_byte_size;
-    uint32_t input_buffer_count;
-    GUARDED_RESPOND_IF_ERROR(
-        responses, r,
-        TRITONBACKEND_InputProperties(
-            input, nullptr /* input_name */, &input_datatype, &input_shape,
-            &input_dims_count, &input_byte_size, &input_buffer_count));
-    if (responses[r] == nullptr) {
-      LOG_MESSAGE(
-          TRITONSERVER_LOG_ERROR,
-          (std::string("request ") + std::to_string(r) +
-           ": failed to read input properties, error response sent")
-              .c_str());
-      continue;
-    }
-
-    LOG_MESSAGE(
-        TRITONSERVER_LOG_INFO,
-        (std::string("\tinput ") + input_name +
-         ": datatype = " + TRITONSERVER_DataTypeString(input_datatype) +
-         ", shape = " + backend::ShapeToString(input_shape, input_dims_count) +
-         ", byte_size = " + std::to_string(input_byte_size) +
-         ", buffer_count = " + std::to_string(input_buffer_count))
-            .c_str());
-    LOG_MESSAGE(
-        TRITONSERVER_LOG_INFO,
-        (std::string("\trequested_output ") + requested_output_name).c_str());
-
-    // For statistics we need to collect the total batch size of all
-    // the requests. If the model doesn't support batching then each
-    // request is necessarily batch-size 1. If the model does support
-    // batching then the first dimension of the shape is the batch
-    // size.
-    if (supports_batching && (input_dims_count > 0)) {
-      total_batch_size += input_shape[0];
-    } else {
-      total_batch_size++;
-    }
-
-    // We only need to produce an output if it was requested.
-    if (requested_output_count > 0) {
-      // This backend simply copies the input tensor to the output
-      // tensor. The input tensor contents are available in one or
-      // more contiguous buffers. To do the copy we:
-      //
-      //   1. Create an output tensor in the response.
-      //
-      //   2. Allocate appropriately sized buffer in the output
-      //      tensor.
-      //
-      //   3. Iterate over the input tensor buffers and copy the
-      //      contents into the output buffer.
-      TRITONBACKEND_Response* response = responses[r];
-
-      // Step 1. Input and output have same datatype and shape...
-      TRITONBACKEND_Output* output;
+      TRITONBACKEND_Input* input = nullptr;
       GUARDED_RESPOND_IF_ERROR(
           responses, r,
-          TRITONBACKEND_ResponseOutput(
-              response, &output, requested_output_name, input_datatype,
-              input_shape, input_dims_count));
+          TRITONBACKEND_RequestInput(request, input_name, &input));
```

Comment:
You can get the input directly by using `TRITONBACKEND_RequestInputByIndex`, that way you need to get the input name via `TRITONBACKEND_InputProperties` at line 718, but you save one API call